web/src/engine/src/core-processor/coreKeyboardProcessor.ts

```diff
@@ -0,0 +1,144 @@
+  public baseLayout: string;
+
+  public get activeKeyboard(): Keyboard {
+    return this.keyboardInterface.activeKeyboard;
+  }
+
+  public set activeKeyboard(keyboard: Keyboard) {
+    this.keyboardInterface.activeKeyboard = keyboard;
+  }
+
+  get keyboardInterface(): KeyboardMinimalInterface {
+    return this._keyboardInterface;
+  }
+
+  public get layerStore(): MutableSystemStore {
+    // TODO-web-core: link to .kmx layer store
+    return this._layerStore;
+  }
+
+  public get newLayerStore(): MutableSystemStore {
+    // TODO-web-core: link to .kmx new-layer store
+    return this._newLayerStore;
+  }
+
+  public get oldLayerStore(): MutableSystemStore {
+    // TODO-web-core: link to .kmx old-layer store
+    return this._oldLayerStore;
+  }
+
+  public get layerId(): string {
+    return this._layerStore.value;
+  }
+  public set layerId(value: string) {
+    this._layerStore.set(value);
+  }
+
+  public processPostKeystroke(device: DeviceSpec, textStore: TextStore): ProcessorAction {
+    // TODO-web-core: Implement this method
+    return null;
+  }
+
+  public processKeystroke(keyEvent: KeyEvent, textStore: TextStore): ProcessorAction {
+    const preInput = SyntheticTextStore.from(textStore, true);
+    const activeKeyboard = this.activeKeyboard as KMXKeyboard;
+    const status = KM_Core.instance.process_event(activeKeyboard.state, keyEvent.Lcode, keyEvent.Lmodifiers, 1, 0); // TODO-web-core: properly set keyDown and flags
+    if (status != KM_CORE_STATUS.OK) {
+      console.error('KeymanWeb: km_core_process_event failed with status: ' + status);
+      return null;
+    }
+    const processorAction = new ProcessorAction();
+    const core_actions = KM_Core.instance.state_get_actions(activeKeyboard.state);
+
+    textStore.deleteCharsBeforeCaret(core_actions.code_points_to_delete);
+    textStore.insertTextBeforeCaret(core_actions.output);
+    processorAction.beep = core_actions.do_alert;
+    processorAction.triggerKeyDefault = core_actions.emit_keystroke;
+
+    // TODO-web-core: Implement options
+    // process_persist_action(engine, actions->persist_options);
+    // TODO-web-core: do we have to do anything with the new_caps_lock_state?
+    // process_capslock_action(actions->new_caps_lock_state);
+
+    processorAction.transcription = textStore.buildTranscriptionFrom(preInput, keyEvent, false);
+
+    return processorAction;
+  }
+
+  /**
+   * Select the OSK's next keyboard layer based upon layer switching keys as a default
+   * The next layer will be determined from the key name unless otherwise specifed
+   *
+   *  @param  {string}                    keyName     key identifier
+   *  @return {boolean}                               return true if keyboard layer changed
+   */
+  public selectLayer(keyEvent: KeyEvent): boolean {
+    // TODO-web-core: Implement this method
+    return false;
+  }
+
+  // Returns true if the key event is a modifier press, allowing keyPress to return selectively
+  // in those cases.
```

Comment:
We should use the correct jsdoc format here; also should start off on the right foot by adding header comments for all implemented functions (bare getters/setters can probably be skipped)